cmd/agent/api/internal/agent/agent.go

```diff
@@ -87,6 +93,7 @@ func SetupHandlers(
 	r.HandleFunc("/secrets", secretInfo).Methods("GET")
 	r.HandleFunc("/metadata/{payload}", func(w http.ResponseWriter, r *http.Request) { metadataPayload(w, r, hostMetadata, invAgent) }).Methods("GET")
 	r.HandleFunc("/diagnose", func(w http.ResponseWriter, r *http.Request) { getDiagnose(w, r, senderManager) }).Methods("POST")
+	r.HandleFunc("/dump-dogstatsd-contexts", func(w http.ResponseWriter, r *http.Request) { dumpDogstatsdContexts(w, r, demux) }).Methods("POST")
```

Comment:
Minor feedback: actions/verbs are by convention usually not part of URL paths so `/dogstatsd-contexts` would probably be more appropriate for the endpoint.